integration_tests/scripts/ituploader.js

```diff
@@ -0,0 +1,75 @@
+const {ethers, Contract} = require("ethers");
+const crypto = require('crypto');
+const {EthStorage} = require("ethstorage-sdk");
+const core = require('@actions/core');
+const fs = require('fs');
+
+const dotenv = require("dotenv")
+dotenv.config()
+const privateKey = process.env.ES_NODE_UPLOADER_PRIVATE_KEY;
+const contractAddr = process.env.ES_NODE_CONTRACT_ADDRESS;
+const RPC = 'http://65.109.20.29:8545';
+const contractABI = [
+    "function lastKvIdx() public view returns (uint40)"
+]
+
+const provider = new ethers.JsonRpcProvider(RPC);
+const contract = new Contract(contractAddr, contractABI, provider);
+const MAX_BLOB = 256n;
+
+async function UploadBlobsForIntegrationTest() {
+    // put blobs
+    console.log(contractAddr)
+    const es = await EthStorage.create({
+        rpc: RPC,
+        privateKey,
+        address: contractAddr
+    })
+    while (true) {
+        const currentIndex = await contract.lastKvIdx();
+        const totalCount = MAX_BLOB - currentIndex;
+        console.log("Current Number:", currentIndex, " Total Number:", totalCount);
+        if (totalCount <= 0) {
+            break;
+        }
+        const buf = crypto.randomBytes(126976);
+        const cost = await es.estimateCost(buf.subarray(0,32).toString('hex'), buf);
+        console.log(cost);
+
+        // write
+        let status = await es.write(buf.subarray(0,32).toString('hex'), buf);
+        console.log(status);
+        fs.writeFileSync(".data", buf.toString('hex')+'\n', { flag: 'a+' });
+    }
+
+    let latestBlock
+    try {
+        latestBlock = await provider.getBlock();
+        console.log("latest block number is", latestBlock.number);
+    } catch (e) {
+        core.setFailed(`EthStorage: get latest block failed with message: ${e.message}`);
+        return
+    }
+
+    // wait for blobs finalized
```

Comment:
Why do we need to wait for the blobs to be finalized?

Reply:
the bootnode only downloads and writes to files for finalized blobs, otherwise, those blobs only saved in memory and cannot sync to peer.

Reply:
Just wondering how to add the case of mining from the downloader cache...

Reply:
Is that also a potential case we should consider adding to the integration test?

Reply:
Why did we resolve this issue?